xbmc/addons/AddonStatusHandler.cpp

```diff
@@ -100,10 +102,10 @@ void CAddonStatusHandler::Process()
     pDialog->SetLine(1, CVariant{24074});
     pDialog->Open();
 
-    CAddonMgr::GetInstance().GetCallbackForType(m_addon->Type())->RequestRestart(m_addon, true);
+    CServiceBroker::GetBinaryAddonManager().OnRequestRestartEvent(m_addon->ID());
   }
   /* Some required settings are missing/invalid */
-  else if ((m_status == ADDON_STATUS_NEED_SETTINGS) || (m_status == ADDON_STATUS_NEED_SAVEDSETTINGS))
```

Comment:
No, was only used from Visualization addons where already changed.
The old way: https://github.com/AlwinEsch/visualization.shadertoy/blob/c80118548589826f43938de0bb8be25cbf9ed728/src/maindx.cpp#L1110-L1142

Can be everything done with e.g. "kodi::SetSettingInt("lastpresetidx", lastIndex);"

(moved to right place :-) )